osu.Game/Rulesets/Mods/ModCinema.cs

```diff
@@ -3,15 +3,35 @@
 
 using osu.Framework.Graphics.Sprites;
 using osu.Game.Graphics;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.UI;
+using osu.Game.Screens.Play;
 
 namespace osu.Game.Rulesets.Mods
 {
-    public class ModCinema : ModAutoplay
+    public abstract class ModCinema<T> : ModCinema, IApplicableToDrawableRuleset<T>
+        where T : HitObject
+    {
+        public virtual void ApplyToDrawableRuleset(DrawableRuleset<T> drawableRuleset)
+        {
+            drawableRuleset.SetReplayScore(CreateReplayScore(drawableRuleset.Beatmap));
```

Comment:
What about using `IApplicableFailOverride.AllowFail = false` instead?

Reply:
The inheritance chain should mean it already takes the settings from the Autoplay mod. `ModCinema` should inherit `IApplicableFailOverride` settings [as seen here](https://github.com/ppy/osu/blob/8da65fc9094282a458662d32c4aeb55deccfa721/osu.Game/Rulesets/Mods/ModAutoplay.cs#L21)